java/code/src/com/redhat/rhn/manager/recurringactions/RecurringActionManager.java

```diff
@@ -0,0 +1,137 @@
+     * @param taskomaticApiIn the {@link TaskomaticApi}
+     */
+    public static void setTaskomaticApi(TaskomaticApi taskomaticApiIn) {
+        taskomaticApi = taskomaticApiIn;
+    }
+    /**
+     * Singleton Instance to get manager object
+     * @return an instance of the manager
+     */
+    public static RecurringActionManager getInstance() {
+        return MANAGER;
+    }
+
+    /**
+     * Private constructor.
+     */
+    private RecurringActionManager() {
+    }
+
+    /**
+     * Create a new minion recurring action
+     *
+     * @param minionId id of the minion
+     * @param cron the cron expression
+     * @param testMode if test mode
+     * @param active is is active
+     * @param user the user
+     * @throws TaskomaticApiException if TaskoSchedule creation fails
+     */
+    public static void createMinionRecurringAction(long minionId, String cron, boolean testMode,
+                                                   boolean active, User user) throws TaskomaticApiException {
+        if (!SystemManager.isAvailableToUser(user, minionId)) {
+            throw new PermissionException("Minion not accessible to user");
+        }
+        MinionServer minion = MinionServerFactory.lookupById(minionId)
+                .orElseThrow(() -> new EntityNotExistsException(MinionServer.class, minionId));
+        MinionRecurringAction action = new MinionRecurringAction(testMode, active, minion);
+        RecurringActionFactory.save(action);
+
+        taskomaticApi.scheduleSatBunch(user, action.computeTaskoScheduleName(), "recurring-state-apply-bunch", cron);
+    }
+
+    /**
+     * List minion recurring action with minion id
+     *
+     * @param minionId id of the minion
+     * @param user the user
+     * @return list of minion recurring actions
+     */
+    public static List<MinionRecurringAction> listMinionRecurringAction(long minionId, User user) {
+        if (!SystemManager.isAvailableToUser(user, minionId)) {
+            throw new PermissionException("Minion not accessible to user");
+        }
+        return RecurringActionFactory.listMinionRecurringActions(minionId);
+    }
+
+    /**
+     * List group recurring action with group id
+     *
+     * @param groupId id of the group
+     * @param user the user
+     * @return list of group recurring actions
+     */
+    public static List<GroupRecurringAction> listGroupRecurringAction(long groupId, User user) {
+        ServerGroupManager groupManager = ServerGroupManager.getInstance();
+        if (!user.hasRole(RoleFactory.SYSTEM_GROUP_ADMIN)) {
+            throw new PermissionException("User does not have access to group");
+        }
+        try {
+            groupManager.lookup(groupId, user);
+            return RecurringActionFactory.listGroupRecurringActions(groupId);
+        }
+        catch (LookupException e) {
+            throw new PermissionException("User does not have access to group");
```

Comment:
You could pass the `cause` exception to the newly created `PermissionException` as follows:

`PermissionException("User does not have access to group", e);`